observation_portal/proposals/models.py

```diff
@@ -99,8 +99,11 @@ def current_semester(self):
 
     @cached_property
     def current_allocation(self):
+        return self.allocation(semester=self.current_semester)
+
+    def allocation(self, semester):
```

Comment:
This method is used by both the `current_allocation` property, and also in one of the new endpoints to get the allocation for a semester that is not the current one.